src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/incentives/IncentivesService.kt

```diff
@@ -365,6 +415,21 @@ class IncentivesService(
     log.info("Prison Incentive level deleted: $prisonId")
   }
 
+  fun getPrisonIncentiveLevel(prison: String, code: String): PrisonIncentiveLevelDataResponse {
+    // it is possible for Prison Incentive Level to exist without VisitAllowanceLevel but not vice versa
+    val location = agencyLocationRepository.findById(prison)
+      .orElseThrow(BadDataException("Prison with id=$prison does not exist"))
```

Comment:
Rest purist would say this should be a 404 since prison ID is in the resource URL